src/Service.Host/client/src/components/__tests__/supplierUtility.js/Supplier.test.js

```diff
@@ -0,0 +1,506 @@
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        const state = {
+            accountingCompanies: { items: [] },
+            supplier: {
+                loading: false,
+                editStatus: 'edit',
+                item: {
+                    ...supplierData,
+                    links: [
+                        { rel: 'self', href: '/supplier/123' },
+                        { rel: 'edit', href: '/supplier/123' }
+                    ]
+                }
+            }
+        };
+        useSelector.mockImplementation(callback => callback(state));
+        render(<Supplier />);
+
+        const phoneNumberInput = screen.getByLabelText('Phone Number');
+        fireEvent.change(phoneNumberInput, { target: { value: '0987654321' } });
+    });
+
+    test('shoud set edit status...', () => {
+        expect(setSupplierEditStatusSpy).toBeCalledWith('edit');
+    });
+});
+
+describe('When updating supplier...', () => {
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        const state = {
+            accountingCompanies: { items: [] },
+            supplier: {
+                loading: false,
+                editStatus: 'edit',
+                item: {
+                    ...supplierData,
+                    links: [
+                        { rel: 'self', href: '/supplier/123' },
+                        { rel: 'edit', href: '/supplier/123' }
+                    ]
+                }
+            }
+        };
+        useSelector.mockImplementation(callback => callback(state));
+        render(<Supplier />);
+
+        const phoneNumberInput = screen.getByLabelText('Phone Number');
+        fireEvent.change(phoneNumberInput, { target: { value: '0987654321' } });
+        const saveButton = screen.getByRole('button', { name: 'Save' });
+        fireEvent.click(saveButton);
+    });
+
+    test('shoud clear errors dispatch update action...', () => {
+        expect(clearSupplierErrorsSpy).toBeCalledTimes(1);
+        expect(updateSupplierSpy).toBeCalledWith(
+            123,
+            expect.objectContaining({
+                ...supplierData,
+                phoneNumber: '0987654321'
+            })
+        );
+    });
+});
+
+describe('When creating supplier...', () => {
+    beforeEach(async () => {
+        jest.clearAllMocks();
+        const state = {
+            accountingCompanies: { items: [] },
+            supplier: {
+                loading: false,
+                editStatus: 'edit'
+            }
+        };
+        useSelector.mockImplementation(callback => callback(state));
+        render(<Supplier creating />);
+
+        const nameInput = screen.getByLabelText('Phone Number');
+        fireEvent.change(nameInput, { target: { value: 'NEW SUPPLIER NAME' } });
+        const saveButton = screen.getByRole('button', { name: 'Save' });
+        fireEvent.click(saveButton);
+    });
+
+    test('shoud clear errors dispatch add action...', () => {
```

Comment:
doesn't matter but missing an 'L' in should here and the one above lol, also maybe an 'and', but really commenting because for most they're "should do x" but there's a few above that don't follow that like 'Page dispatch put on hold action'. Obviously doesn't stop ya putting it through but could make them all 'should blah' for the next one